extensions/wikia/UserRenameTool/process/ProcessBase.class.php

```diff
@@ -0,0 +1,556 @@
+		if ( empty( $params ) ) {
+			return;
+		}
+
+		$renameData = $this->getRenameData( $user );
+		$renameData = array_merge( $renameData, $params );
+		$this->setRenameData( $user, $renameData );
+	}
+
+	/**
+	 * Performs action for cleaning up temporary data at the very end of a process
+	 */
+	public function cleanupFakeUser() {
+		if ( $this->fakeUserId ) {
+			$this->logInfo( "Writing renameData for fake user" );
+
+			$fakeUser = \User::newFromId( $this->fakeUserId );
+
+			$this->setRenameData( $fakeUser, [ self::RENAME_TAG => $this->newUsername ] );
+			$fakeUser->saveSettings();
+			$fakeUser->saveToCache();
+		}
+	}
+
+	/**
+	 * Log when the rename operation starts to the staff log
+	 *
+	 * @param string $taskId
+	 */
+	public function logStartToStaff( $taskId = null ) {
+		$logLink = $taskId ? \UserRenameToolHelper::buildTaskLogLink( $taskId ) : '-';
+		$this->addStaffLogAction( self::ACTION_START, 'userrenametool-info-started', $logLink );
+	}
+
+	/**
+	 * Log when the rename operation has failed to the staff log
+	 *
+	 * @param string $taskId
+	 */
+	public function logFailToStaff( $taskId = null ) {
+		$logLink = $taskId ? \UserRenameToolHelper::buildTaskLogLink( $taskId ) : '-';
+		$this->addStaffLogAction( self::ACTION_FAIL, 'userrenametool-info-failed', $logLink );
+	}
+
+	/**
+	 * Log when the rename operation has completed to the staff log
+	 *
+	 * @param string $taskId
+	 */
+	public function logFinishToStaff( $taskId = null ) {
+		$logLink = $taskId ? \UserRenameToolHelper::buildTaskLogLink( $taskId ) : '-';
+		$this->addStaffLogAction( self::ACTION_FINISH, 'userrenametool-info-finished', $logLink );
+	}
+
+	/**
+	 * Log a successful rename for a single wiki
+	 *
+	 * @param int $wikiId The wiki ID for this single rename.  If not given the current value of wgCityId is used.
+	 */
+	public function logFinishWikiToStaff( $wikiId = null ) {
+		$wikiId = $wikiId ? $wikiId : \F::app()->wg->CityId;
+		$this->addStaffLogAction( self::ACTION_LOG, 'userrenametool-info-wiki-finished', $wikiId );
+	}
+
+	/**
+	 * Log a rename fail for a single wiki
+	 *
+	 * @param int $wikiId The wiki ID for this single rename.  If not given the current value of wgCityId is used.
+	 */
+	public function logFailWikiToStaff( $wikiId = null ) {
+		$wikiId = $wikiId ? $wikiId : \F::app()->wg->CityId;
+		$this->addStaffLogAction( self::ACTION_LOG, 'userrenametool-info-wiki-finished-problems', $wikiId );
+	}
+
+	public function addStaffLogAction( $action, $key, $info = null ) {
+		$formattedLogLine = \UserRenameToolHelper::generateLogLine(
+			$key,
+			$this->requestorName,
+			$this->oldUsername,
+			$this->newUsername,
+			$this->reason,
+			$info
+		);
+
+		$this->addStaffLog( $action, $formattedLogLine );
+	}
+
+	/**
+	 * Logs the message to main user-visible log
+	 *
+	 * @param string $action
+	 * @param $text string Log message
+	 */
+	public function addStaffLog( $action, $text ) {
+		\StaffLogger::log(
+			'renameuser',
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "renameuser" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)